catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -883,11 +878,11 @@ private int columnPosition(List<FieldSchema> columns, TableChange.ColumnPosition
       int indexOfColumn = indexOfColumn(columns, afterColumn);
       Preconditions.checkArgument(indexOfColumn != -1, "Column does not exist: " + afterColumn);
       return indexOfColumn + 1;
-    } else if (position instanceof TableChange.First) {
-      return 0;
     } else {
-      throw new UnsupportedOperationException(
+      Preconditions.checkArgument(
```

Comment:
I found that some of the changes here and below change the semantics (change to the different exception), I would  suggest we don't do this change and keep as it was.